synapse/handlers/receipts.py

```diff
@@ -111,6 +111,9 @@ async def _handle_new_receipts(self, receipts: List[ReadReceipt]) -> bool:
             # no new receipts
             return False
 
+        if not self.hs.should_send_read_receipts():
```

Comment:
I think this is bailing out too early. We need to make sure that the pushers are notified of new receipts, so that notifications can be removed when messages are read on other servers.

Reply:
You're right. I don't think it should be bailing out at all, should it...